apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -1058,4 +1059,35 @@ export class ApplicationController {
       throw new NotFoundException('Attachment not found')
     }
   }
+
+  @Scopes(ApplicationScope.write)
+  @Delete('applications/:id')
+  @ApiParam({
+    name: 'id',
+    type: String,
+    required: true,
+    description: 'The id of the application to delete.',
+    allowEmptyValue: false,
+  })
+  async delete(
+    @Param('id', new ParseUUIDPipe()) id: string,
+    @CurrentUser() user: User,
+  ) {
+    const { nationalId } = user
+    const existingApplication = (await this.applicationAccessService.findOneByIdAndNationalId(
```

Comment:
Should we try to return early here if we know the application doesnt exist for the specific user?